python/graphframes/tests.py

```diff
@@ -204,3 +237,67 @@ def test_triangle_counts(self):
         c = g.triangleCount()
         for row in c.select("id", "count").collect():
             self.assertEqual(row.asDict()['count'], 1)
+
+class GraphFrameExamplesTest(GraphFrameTestCase):
+    def setUp(self):
+        super(GraphFrameExamplesTest, self).setUp()
+        self.sqlContext = self.sql
+        self.japi = _java_api(self.sqlContext._sc)
+
+    def test_belief_propagation(self):
+        # create graphical model g of size 3 x 3
+        g = Graphs(self.sqlContext).gridIsingModel(3)
+
+        # run BP for 5 iterations
+        numIter = 5
+        results = BeliefPropagation.runBPwithGraphFrames(g, numIter)
+
+        # check beliefs are valid
+        for row in results.vertices.select('belief').collect():
+            belief = row['belief']
+            self.assertTrue(
+                0 <= belief <= 1,
+                msg="Expected belief to be probability in [0,1], but found {}".format(belief))
+
+    def test_graph_friends(self):
```

Comment:
This test doesn't feel very informative. The "fiends" are hardcoded in both the example and the test so this test is mostly testing if someone has changed the package source.